src/Tokenizer/Tokens.php

```diff
@@ -869,7 +869,27 @@ public function findSequence(array $sequence, $start = 0, $end = null, $caseSens
     public function insertAt($index, $items)
     {
         $items = \is_array($items) || $items instanceof self ? $items : [$items];
-        $itemsCnt = \count($items);
+
+        $this->insertSlices([$index => $items]);
+    }
+
+    /**
+     * @TODO: docs
+     *
+     * @internal
+     *
+     * 5 => tokenA
+     * 7 => [tokenB, tokenC]
+     *
+     * @param mixed $slices
```

Comment:
`array<int, Token>`?

Reply:
If I look at the description it should be `array<int, Token|Token[]>`

Reply:
indeed, i simply didn't documented this methods (as I put todo), as i wanted the idea to be validated first ;)